code/game/machinery/doors/brig_system.dm

```diff
@@ -9,350 +15,346 @@
 
-	for(var/obj/item/paper/incident/I in incident_reports)
-		var/datum/crime_incident/incident = I.incident
+		data["incidents"] += list(incident_data)
 
-		if(!incident.active_timer)
-			continue
+	// Does this user have the ability to pardon?
+	data["can_pardon"] = FALSE
+	if (ishuman(user))
+		var/mob/living/carbon/human/human = user
+		var/obj/item/card/id/id_card = human.get_idcard()
 
-		if(REALTIMEOFDAY > incident.time_to_release)
-			timer_end(I)
+		if (id_card)
+			if ((id_card.paygrade in GLOB.co_paygrades) || (id_card.paygrade in GLOB.highcom_paygrades) || (id_card.paygrade == "PvI"))
+				data["can_pardon"] = TRUE
 
-	updateUsrDialog()
-	update_icon()
+	return data
 
-/obj/structure/machinery/brig_cell/update_icon()
-	if(stat & (NOPOWER))
-		icon_state = "frame"
-		return
 
-	if(stat & (BROKEN))
-		set_picture("ai_bsod")
-		return
+/obj/structure/machinery/brig_cell/ui_static_data()
+	var/list/data = list()
 
-	if(current_report && current_report.incident.active_timer)
-		var/disp1 = id
-		var/time_left = get_time_left(current_report)
-		var/disp2 = "[add_zero(num2text((time_left / 60) % 60),2)]~[add_zero(num2text(time_left % 60), 2)]"
-		if(length(disp2) > 5)
-			disp2 = "Error"
-		update_display(disp1, disp2)
-	else
-		if(maptext)
-			maptext = ""
+	// Bitflags.
+	data["bit_active"] = BRIG_SENTENCE_ACTIVE
+	data["bit_served"] = BRIG_SENTENCE_SERVED
+	data["bit_pardoned"] = BRIG_SENTENCE_PARDONED
+	data["bit_perma"] = BRIG_SENTENCE_PERMA
 
-/obj/structure/machinery/brig_cell/power_change()
-	..()
+	return data
 
-	update_icon()
 
-/obj/structure/machinery/brig_cell/attackby(var/obj/item/W, var/mob/living/user)
-	if(!istype(W, /obj/item/paper/incident))
-		return
+/obj/structure/machinery/brig_cell/proc/get_time_left(datum/crime_incident/incident)
+	var/time = (incident.time_to_release - REALTIMEOFDAY)/10
+
+	if(!(incident.status & BRIG_SENTENCE_ACTIVE))
+		time = ((incident.brig_sentence * 600) - incident.time_served)/10
+	return max(0, time)
 
-	var/obj/item/paper/incident/I = W
-	if(!I.incident || !I.incident.brig_sentence || I.incident.pardoned)
+
+/obj/structure/machinery/brig_cell/ui_act(action, params)
+	. = ..()
+	if(.)
 		return
 
-	user.temp_drop_inv_item(I)
-	I.moveToNullspace()
-	incident_reports += I
+	switch (action)
+		if ("set_viewed_incident")
+			var/new_incident = params["incident"]
+			if (new_incident)
+				viewed_report = locate(new_incident)
+			else
+				viewed_report = null
+
+		if ("toggle_doors")
+			for (var/obj/structure/machinery/door/poddoor/almayer/locked/door in targets)
+				if (door.density)
+					door.open()
+				else
+					door.close()
+
+		if ("flash")
+			for (var/obj/structure/machinery/flasher/flasher in targets)
+				flasher.flash()
+
+		if ("start_timer")
+			// Pause existing timer, if there is one.
+			timer_pause(usr)
+			timer_start(usr)
+
+		if ("pause_timer")
+			timer_pause(usr)
+
+		if ("end_timer")
+			timer_end(usr)
+
+		if ("reset_timer")
+			timer_reset(usr)
+
+		if ("pardon")
```

Comment:
check for if they can pardon. i know the button appears dependent on this - but you can exploit tgui to send whatever ui_acts you like